### PR TITLE
parser: implement Restore for DropIndexStmt

### PR DESCRIPTION
**Pull request by honestold3**
implement DropIndexStmt#Restore

add ddl_test.go#TestDropIndex

https://github.com/pingcap/tidb/issues/8532

### CODE REVIEW

File: ast/ddl.go [hunk 1]
```diff
@@ -752,7 +752,17 @@ type DropIndexStmt struct {
...
-	return errors.New("Not implemented")
+	ctx.WriteKeyWord("DROP INDEX ")
+	if n.IfExists {
+		ctx.WriteKeyWord("IF EXISTS ")
+	}
+
+	if err := n.Table.Restore(ctx); err != nil {
+		return errors.Annotate(err, "An error occurred while splicing IndexHints")
+	}
+	ctx.WritePlain(" on ")
```
Comment (kennytm) [inline ast/ddl.go]: This should be a WriteKeyWord

**Comment by leoppro** [comment]
thanks for your contribution!

File: ast/ddl_test.go [hunk 2]
```diff
@@ -62,3 +62,17 @@ func (ts *testDDLSuite) TestDDLVisitorCover(c *C) {
 		v.node.Accept(visitor1{})
 	}
 }
+
+func (ts *testDDLSuite) TestDropIndex(c *C) {
```
[the remark this answers is not in the document]
Comment (leoppro) [reply ast/ddl_test.go]: see: https://github.com/pingcap/blog-cn/pull/137/files?utf8=%E2%9C%93&diff=unified#diff-5272326fe89eb3997d1363d1b21df828R233

File: ast/ddl_test.go [hunk 3]
```diff
@@ -62,3 +62,18 @@ func (ts *testDDLSuite) TestDDLVisitorCover(c *C) {
...
+	testCases := []NodeRestoreTestCase{
```
Comment (leoppro) [inline ast/ddl_test.go]: please see:parser_test.go#RunTest and parser_test.go#RunRestoreTest
and see parser_test.go#L1406-L1408
```
		{"use `select`", true, "USE `select`"},
		{"use `sel``ect`", true, "USE `sel``ect`"},
		{"use select", false, "USE `select`"},
```

we hope you can test `DropIndexStmt` kile `UseStmt` in `parser_test.go`

File: parser_test.go [hunk 4]
```diff
@@ -1784,6 +1784,14 @@ func (s *testParserSuite) TestDDL(c *C) {
 		{"CREATE INDEX idx USING BTREE ON t (a) USING HASH COMMENT 'foo'", true, ""},
 		{"CREATE INDEX idx USING BTREE ON t (a)", true, ""},
 
+		//For dorp index statement
+		{"DROP INDEX `a` ON `t`", true, ""},
```
Comment (leoppro) [inline parser_test.go]: the left string is test data, the right sting is expected text. test it like this:
```suggestion
		{"DROP INDEX a ON t", true, "DROP INDEX `a` ON `t`"},
```

File: ast/ddl_test.go [hunk 5]
```diff
@@ -62,3 +62,18 @@ func (ts *testDDLSuite) TestDDLVisitorCover(c *C) {
 		v.node.Accept(visitor1{})
 	}
 }
+
+func (ts *testDDLSuite) TestDropIndex(c *C) {
```
Comment (leoppro) [inline ast/ddl_test.go]: you can remove it now

**Comment by leoppro** [comment]
LGTM, @kennytm PTAL

**Pull request merged**